### PR TITLE
Enable customizing allowed labels in hook plugin

### PR DESCRIPTION
**Pull request by kargakis**
Fixes https://github.com/kubernetes/test-infra/issues/7929

/assign @cjwagner @stevekuznetsov

### CODE REVIEW

File: prow/plugins/label/label.go [hunk 1]
```diff
@@ -78,7 +78,36 @@ func getLabelsFromREMatches(matches [][]string) (labels []string) {
 	return
 }
 
-func handle(gc githubClient, log *logrus.Entry, e *github.GenericCommentEvent) error {
+func getLabelsFromGenericMatches(matches [][]string, allowedLabels []plugins.Label) (labels []string) {
+	if len(allowedLabels) == 0 {
+		return nil
+	}
+out:
```
Comment (stevekuznetsov) [inline prow/plugins/label/label.go]: Prefer functions and `return ` over labels?

File: prow/plugins/label/label.go [hunk 2]
```diff
@@ -78,7 +78,36 @@ func getLabelsFromREMatches(matches [][]string) (labels []string) {
 	return
 }
 
-func handle(gc githubClient, log *logrus.Entry, e *github.GenericCommentEvent) error {
+func getLabelsFromGenericMatches(matches [][]string, allowedLabels []plugins.Label) (labels []string) {
```
Comment (stevekuznetsov) [inline prow/plugins/label/label.go]: Doc this please

File: prow/plugins/label/label.go [hunk 3]
```diff
@@ -78,7 +78,36 @@ func getLabelsFromREMatches(matches [][]string) (labels []string) {
...
+	if len(allowedLabels) == 0 {
+		return nil
+	}
+out:
+	for _, match := range matches {
+		parts := strings.Split(match[0], " ")
```
Comment (stevekuznetsov) [inline prow/plugins/label/label.go]: Can't we add a capturing group for the command and not need this, feels like the input here should already be processed

Comment (kargakis) [reply prow/plugins/label/label.go]: We already process labels like this so I prefer not to fix it here.

File: prow/plugins/label/label.go (hunk 1 above)
Comment (kargakis) [reply prow/plugins/label/label.go]: I have simplified the plugin config which obsoletes the need for this.

File: prow/plugins/label/label.go (hunk 2 above)
Comment (kargakis) [reply prow/plugins/label/label.go]: done

**Pull request merged**